lib/chef/resource/build_essential.rb

```diff
@@ -45,24 +45,29 @@ class BuildEssential < Chef::Resource
         description: "Install the build essential packages at compile time.",
         default: false, desired_state: false
 
+      property :raise_if_unsupported, [TrueClass, FalseClass],
+        description: "Raise a hard error on platforms where this resource is unsupported.",
+        default: false, desired_state: false # FIXME: make this default to true
```

Comment:
making the default to continue on when we don't do what the user says seems like a really big anti-pattern. if that's what the user wants then they should specify it. Shouldn't be the default

Reply:
Its the current behavior, it would likely be breaking to change it